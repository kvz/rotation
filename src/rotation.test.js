const rotation = require('./rotation')
const isodate = require('./isodate')

describe('rotation', () => {
  test('should match snapshot for every day from 2017-01-01 to 2019-01-01', () => {
    let msgs = []
    for (let i = 1483232461; i <= 1546304461; i = i + 86400) {
      let d = new Date(i * 1000)
      let res = rotation(d)
      let id = isodate(d)
      let msg = `${id}: ${res}`
      msgs.push(msg)
    }
    expect(msgs).toMatchSnapshot()
    // expect(rotation(new Date('2017-05-01T15:01:01'))).toBe('saturday')
    // expect(rotation(new Date('2017-05-06T15:01:01'))).toBe('saturday')
    // expect(rotation(new Date('2017-05-07T15:01:01'))).toBe('sunday')
    // expect(rotation(new Date('2017-05-08T15:01:01'))).toBe('monday')
  })
})
