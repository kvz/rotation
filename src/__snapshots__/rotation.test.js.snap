// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rotation should match snapshot for every day from 2017-01-01 to 2019-01-01 1`] = `
Object {
  "daily-0-sunday": "2018-12-30",
  "daily-1-monday": "2018-12-31",
  "daily-2-tuesday": "2018-12-25",
  "daily-3-wednesday": "2018-12-26",
  "daily-4-thursday": "2018-12-27",
  "daily-5-friday": "2018-11-30",
  "daily-6-saturday": "2018-12-29",
  "monthly-01-january": "2019-01-01",
  "monthly-02-february": "2018-02-01",
  "monthly-03-march": "2018-03-01",
  "monthly-05-may": "2018-05-01",
  "monthly-06-june": "2018-06-01",
  "monthly-07-july": "2018-07-01",
  "monthly-09-september": "2018-09-01",
  "monthly-10-october": "2018-10-01",
  "monthly-11-november": "2018-11-01",
  "quarterly-2017-0": "2017-04-01",
  "quarterly-2017-1": "2017-08-01",
  "quarterly-2017-2": "2017-12-01",
  "quarterly-2018-0": "2018-04-01",
  "quarterly-2018-1": "2018-08-01",
  "quarterly-2018-2": "2018-12-01",
  "weekly-1": "2018-12-07",
  "weekly-2": "2018-12-14",
  "weekly-3": "2018-12-21",
  "weekly-4": "2018-12-28",
}
`;
